src/core/lib/event_engine/default_event_engine.cc

```diff
@@ -37,56 +39,93 @@ namespace grpc_event_engine {
 namespace experimental {
 
 namespace {
-std::atomic<absl::AnyInvocable<std::unique_ptr<EventEngine>()>*>
+std::atomic<absl::AnyInvocable<std::shared_ptr<EventEngine>()>*>
     g_event_engine_factory{nullptr};
 grpc_core::NoDestruct<grpc_core::Mutex> g_mu;
-grpc_core::NoDestruct<std::weak_ptr<EventEngine>> g_event_engine;
+grpc_core::NoDestruct<std::weak_ptr<EventEngine>> g_weak_internal_event_engine;
```

Comment:
Please add lock annotations for these two globals.

Reply:
The warnings show up as you expected, even with NoDestruct `operator*`! Very neat.